Library/Homebrew/formula_specialties.rb

```diff
@@ -18,12 +18,18 @@ def self.url(val)
 # This formula serves as the base class for several very similar
 # formulae for Amazon Web Services related tools.
 class AmazonWebServicesFormula < Formula
+
   # Use this method to peform a standard install for Java-based tools,
   # keeping the .jars out of HOMEBREW_PREFIX/lib
   def install
     rm Dir['bin/*.cmd'] # Remove Windows versions
     libexec.install Dir['*']
-    bin.install_symlink Dir["#{libexec}/bin/*"] - ["#{libexec}/bin/service"]
+    Pathname.glob("#{libexec}/bin/*") do |file|
+      next if file.directory?
+      basename = file.basename
+      next if basename.to_s == "service"
+      (bin/basename).write_env_script file, :JAVA_HOME => ENV["JAVA_HOME"]
```

Comment:
The objects I want to accomplish:
* Users don't need to set JAVA_HOME, we set for them in the shim script.
* If a user upgrade their java say from `1.8.0_31` to `1.8.0_32`, the script can still work. Current `:JAVA_HOME => ENV["JAVA_HOME"]` doesn't qualify. We write the full JDK path into the shim script, like this:
```
#!/bin/bash
JAVA_HOME="/Library/Java/JavaVirtualMachines/jdk1.8.0_31.jdk/Contents/Home" ...
```
This problem applies to all Java formulae which use `write_env_script`. And it's bad for bottle perspective.